hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterImpl.java

```diff
@@ -319,6 +323,9 @@ protected void checkBlockBoundary() throws IOException {
       shouldFinishBlock = blockWriter.encodedBlockSizeWritten() >= hFileContext.getBlocksize()
         || blockWriter.blockSizeWritten() >= hFileContext.getBlocksize();
     }
+    if (blockWriter.isSizeLimitCompressed()) {
```

Comment:
let's change this to `if (blockWriter.isSizeLimitCompressed() && !shouldFinishBlock)`? Just noting your comment on the calculation of compression ratio in the other file, we could further avoid that cost if we already know we need to finish the block for other reasons.